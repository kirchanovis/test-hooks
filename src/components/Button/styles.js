import styled from 'styled-components'

export const
    Button = styled.button`
        position: relative;
        display: inline-block;
        padding: 12px 24px;
        margin: .3em 0 1em 0;
        width: 100%;
        vertical-align: middle;
        color: #fff;
        font-size: 16px;
        line-height: 20px;
        -webkit-font-smoothing: antialiased;
        text-align: center;
        letter-spacing: 1px;
        background: transparent;
        border: 0;
        border-bottom: 2px solid #3160B6;
        cursor: pointer;
        transition: all 0.15s ease;
        background: #4a89dc;
        text-shadow: 1px 1px 0 rgba(39, 110, 204, .5);
        &:hover{
            background: #357bd8;
        }
        &:disabled{
            opacity: 0.4;
            pointer-events: none;
        }
    `;