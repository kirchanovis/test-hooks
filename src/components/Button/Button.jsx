import React from 'react';
import { Button as Root } from './styles'

/**
 * @typedef Props
 */
type Props = {

  text: string,

  disabled?: boolean,

  valid?: boolean
}

function Button(props: Props) {
    const {
        text,
        disabled,
        valid
    } = props

    return (
        <Root
            disabled={disabled || valid}
            type="submit"
        >
            {text}
        </Root>
    );
}

export default Button;