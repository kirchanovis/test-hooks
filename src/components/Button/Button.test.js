import renderer from 'react-test-renderer'
import React from 'react';
import { Button } from './../Button';

test('renders default button', () => {
    const tree = renderer.create(
        <Button
            text="Example"
        />
    ).toJSON();

    expect(tree).toMatchSnapshot();
});

test('renders disabled button', () => {
    const tree = renderer.create(
        <Button
            disabled
            text="Example"
        />
    ).toJSON();

    expect(tree).toMatchSnapshot();
});


