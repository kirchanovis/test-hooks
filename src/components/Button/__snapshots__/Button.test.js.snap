// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders default button 1`] = `
<button
  className="sc-bdVaJa bNwIqc"
  type="submit"
>
  Example
</button>
`;

exports[`renders disabled button 1`] = `
<button
  className="sc-bdVaJa bNwIqc"
  disabled={true}
  type="submit"
>
  Example
</button>
`;
